File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let answer = [];
    let sum = 0;
    let trace = 0;
    let interval = sequence.length;
    for(let i = 0 ;i<sequence.length;i++){
        sum += sequence[i];
        while(sum > k){
            sum -= sequence[trace++];
        }
        if(sum === k){
            answer.push([trace,i]);
        }
    }
    console.log(answer);
    let min = 100000
    let savevalue = 0;
    for(let i = 0;i<answer.length;i++){
        if(answer[i][1] - answer[i][0] < min){
            min = answer[i][1] - answer[i][0];
            savevalue = i;
        }
    }
    return answer[savevalue];
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer(2,0);
    int sum = 0;
    int start = 0;
    int end = 0;
    int save_min = 1e9;
    for(int i = 0;i<sequence.size();i++){
        sum += sequence[i];
        end = i;
        while(sum > k){
            sum -= sequence[start++];
        }
        if(sum == k){
            if(end - start < save_min){
                answer[0] = start;
                answer[1] = end;
                save_min = end - start;
            }
        }
    }
    return answer;
}